core/lib/spree/core/engine.rb

```diff
@@ -74,6 +74,15 @@ class Engine < ::Rails::Engine
             caller
           )
         end
+
+        if Spree::Config.use_legacy_address_state_validator != false
+          Spree::Deprecation.warn(
+            'Spree::Config.use_legacy_address_state_validator set to true is ' \
+            'deprecated and will not be possibile in Solidus 3.0. Please switch this ' \
+            'value to false and check that everything works as expected.',
```

Comment:
Can we improve this message to explain that validation will be done through an external class moving forward?

Reply:
@aldesantis what do you think about this message?

> DEPRECATION WARNING: Spree::Config.use_legacy_address_state_validator set to true has been deprecated and will be removed in Solidus 3.0. The Spree::Address state validation has been extracted into a configurable external class. Switch Spree::Config.use_legacy_address_state_validator to true to start using the external validator class.

Reply:
@cedum sounds great!